File: app/library/GPIO.php
<?php

class GPIO {

    public static $out = 0;
    public static $in = 1;
    public static $board = 10;
    public static $bcm = 11;

    public static function clean() {

        return self::sendCommand('clean');

    }

    public static function setup($channel, $direction) {

        return self::sendCommand('setup '.$channel.' '.$direction);

    }

    public static function output($channel, $state) {

        return self::sendCommand('output '.$channel.' '.$state);

    }

    private function sendCommand($params) {
        return shell_exec('python ' . __DIR__.'/gpio.py '.$params);
    }

}

File: app/library/gpio.py
#!/usr/bin/env python
import RPi.GPIO as GPIO
import sys

if sys.argv[1] == 'clean':
    GPIO.setwarnings(False)
    GPIO.cleanup()
    GPIO.setwarnings(True)
elif sys.argv[1] == 'setup':
    print 'setup'
    GPIO.setmode(GPIO.BCM)
    GPIO.setup(int(sys.argv[2]), int(sys.argv[3]))
elif sys.argv[1] == 'output':
    print 'output'
    GPIO.setmode(GPIO.BOARD)
    GPIO.output(int(sys.argv[2]), int(sys.argv[3]))



